app/src/main/java/org/xbmc/kore/provider/MediaDatabase.java

```diff
@@ -346,6 +354,7 @@ public void onCreate(SQLiteDatabase db) {
                    MediaContract.SyncColumns.UPDATED + " INTEGER NOT NULL," +
                    MediaContract.SongsColumns.HOST_ID + " INTEGER NOT NULL " + References.HOST_ID + ", " +
                    MediaContract.SongsColumns.ALBUMID + " INTEGER NOT NULL, " +
+                   MediaContract.SongsColumns.ARTISTID + " INTEGER NOT NULL, " +
```

Comment:
Probably, but the main issue is that in Kodi the connection between songs and artists is many to many and this model should respect that (see main comment)